app/taxonomy_es.py

```
@@ -104,32 +101,131 @@ def create_synonyms_files(taxonomy: Taxonomy, langs: list[str], target_dir: Path
+        for set_num, batched_synonyms_sets_items in enumerate(
+            batch(lang_synonyms_sets.items(), MAX_SYNONYMS_SETS_RULES)
+        ):
+            # see
+            # https://www.elastic.co/docs/reference/text-analysis/analysis-synonym-graph-tokenfilter#_solr_synonyms_2
+            # for rule definition
+            set_id = f"{set_name}-{set_num}"
+            es.synonyms.put_synonym(
+                id=set_id,
+                synonyms_set=[
+                    {"id": id, "synonyms": f"{','.join(synonyms)} => {id}"}
+                    for id, synonyms in batched_synonyms_sets_items
```

Ruff (0.16.0)

[error] Variable `id` is shadowing a Python builtin

(A001)